src/components/add-transaction-dialog/add-transaction-dialog.js

```diff
@@ -57,19 +96,44 @@ export class AddTransactionComponent {
     this.dialog.listen("MDCDialog:cancel", this.handleCancel.bind(this));
   }
 
+  getType() {
+    return this.income.checked ? "+" : "-";
+  }
+
+  getAccount() {
+    return this.accounts.find(
+      item => this.accountSelect.getValue() === item.name
+    );
+  }
+
   handleOk() {
-    console.log("accepted");
+    this.props.addTransaction({
+      type: this.getType(),
+      date: this.date.value,
+      amount: parseInt(this.amount.value),
+      desc: this.description.value,
+      tag: this.tagSelect.getValue(),
+      account: this.getAccount(),
+      id: Date.now()
+    });
+    this.cleanDialog();
   }
 
   handleCancel() {
-    console.log("declined");
+    this.cleanDialog();
+  }
+
+  cleanDialog() {
+    this.income.checked = true;
```

Comment:
let (by default) expense checkbox will be checked (because people spent money more often than have income :) )